### ISSUE TITLE
Regarding running YOLOP for a custom dataset, runtime warning: invalid value encountered in true_divide

### ISSUE THREAD

**Issue by Aswin-Prasannakumar** [hustvl/YOLOP#225]
I am new to Computer Vision and i was trying to run YOLOP using a custom dataset that i built. i have images of size 400*300 and the detection annotation labels are structured as attached below

[11383.json](https://github.com/user-attachments/files/18475941/11383.json)

I am only training the encoder and detection heads, so, i have put _C.TRAIN.ENC_DET_ONLY = True and everything else to false. But to keep the dataset folders in line with the recommendation i have added other png images of size 400*300 which are completely black (i just made them using python to fill the folders) to the da_seg_annotation and ll_seg_annotation folders. so to summarize, i have actual frames in my images folder and their labels in the json file structured as above, then i have black png images o the same size as the ones in the image folder so that i dont have to modify much in the original code. But now while training i am getting a runtime warning: invalid value encountered in true_divide 
IoU=intersection / union 
and then when it starts to train,Epoch: [1][0/3275]	Time 1488.698s (1488.698s)	Speed 0.0 samples/s	Data 27.704s (27.704s)	Loss nan (nan)
NaN or Inf found in input tensor - is happening

my frames are of the type (this is a random image from my dataset and not associated with the label given above): - 

![Image](https://github.com/user-attachments/assets/bd3629e2-3ebf-44de-8549-7aff75bc84b5)

and the frames which i used to fill the ll_seg_annotations and da_seg_annotations folder are as follows : -

![Image](https://github.com/user-attachments/assets/6fb07767-665f-47e6-b616-46182a9cda40)

could this be an issue that the images are very different even if i am not training the driving area and lane line segmentation heads?

i am also getting a warning that says WARNING: Extremely small objects found. 193 of 111455 labels are < 3 pixels in width or height. could this be related to the runtime warning: invalid value encountered in true_divide 
IoU=intersection / union  warning i am getting and also the speed is 0.0 samples/s and the loss is shown as nan (nan) NaN or Inf found in input tensor issue while training

**Comment by habibanaser** [comment]
did you solve the problem? if yes, can you tell me what you did?

**Comment by Aswin-Prasannakumar** [comment]
> did you solve the problem? if yes, can you tell me what you did?

Yes, I have solved it for my case. For me, I believe it was because I was using CUDA version 10.2 which I think didn't support Automatic Mixed Precision (AMP) natively in PyTorch. So, then I tested the dataset by running it on my CPU and it worked fine. then later I upgraded my CUDA version to 11.8 and ran it on my GPU and all is well now.

Also, I increased the resolution of the images in my dataset and I believe that has caused the WARNING: Extremely small objects found. 193 of 111455 labels are < 3 pixels in width or height to go away.

**Comment by habibanaser** [comment]
thank you for your response, have you published the dataset?

**Comment by Aswin-Prasannakumar** [comment]
> thank you for your response, have you published the dataset?

No I'm sorry the dataset has not yet been published
